### ISSUE TITLE
Binary executable in the repository

### ISSUE THREAD

**Issue by pawosm-arm** [rricharz/Tek4010#21]
The Tek4010 contains binary blob on it's master branch. There's an executable binary file called `tek4010` which needs to be removed, otherwise the `make` command won't build binary for a specific system. Having rebuilt/usable binary causes `git` to print error message whenever I want to pull from upstream.

Note that you already export 6 releases for this project and that's the right place for executable binary files.

**Comment by rricharz** [comment]
Sorry for the inconvenience. This is a very good point, and I have been thinking about how to best handle this since a while.

Initially, this project was just for the Raspberry Pi, and it is most convenient for Raspberry Pi users that a simple "pull" will give them the latest version. No libraries to install, nothing to build.

Now as the project is spreading and more people are using it on other Linux and Unix systems, I will have to package the project in a way which is also convenient for users of other systems. I will do that in a while. In the mean time, probably the easiest is to do the following:
  git checkout tek4010
  git pull
  rm tek4010
  make

**Comment by tvrusso** [comment]
I think it is best if binaries (executables and object files) were not present.  I, too, am mostly building this on systems other than the Pi, and the presence of pi-specific binaries is in the way.

Removing the file and rebuilding after a pull is fine, but the presence of the newly rebuilt binary can interfere with future pulls as well.  Better if they weren't there.

BTW, I have some other modifications that make it easier to build on non-pi, non-linux systems (mostly changing Makefiles to not hard-code gcc, and scripts to not hard-code /bin/bash).  If you ever move to make this repo less tied to the pi by removing binaries, I'll submit a PR.

**Comment by udo-munk** [comment]
You could add a clean target to the makefile, so that 'make clean' will remove the binary. Another option would be be different targets for systems like tek4010-pi, tek4010-linux. Then users can use the prebuild binary appropriate for the system. A 'make all' would build all targets then.

**Comment by rricharz** [comment]
I have done quite some research, but could not find a solution which would
1. Allows Raspberry Pi users with no github experience to continue to use the simple "git pull".
2. Allows users on other systems to use a simple "git pull; make"

The first of these requirements is especially important because if I remove the binary tek4010, many unexperienced users will be very unhappy after their next "git pull" and I have not found a way to pop up a message in this case telling them what to do, unless they read the README.MD again.

In my opinion the best solution for now is:
1. To make new releases earliest every 3 months, unless there is an essential bug fix. Tek4010 is now stable enough so that this is hopefully achievable.
2. Provide a "tek-update" script which does a "git pull" if the system is a Raspberry Pi with the current Debian version, otherwise something like a "rm tek4010; git pull; rm tek4010; make". We have to experiment with the best way to detect whether a build is required. Ideally, this script would also check for the dependency "libgtk-3-dev", or whatever else is required on other systems.
3. Change the instructions for updating in README.md to do a "tek-update", so that at least new users will use "tek-update" (if they read the instructions).

If you agree with this proposal, and are willing to contribute, I will create a branch "other_systems" for testing "tek-update" and fixing the "makefile" and the scripts to work on other systems as well. The problem with such a branch is of course that it can only be merged back easily without rebasing if no changes to the master branch are required in the mean time. Also, in my opinion it is very important to keep the "makefile" and the scripts as simple as possible, so that users with limited experience can understand them. I recently worked on a project in which the "makefile" and the scripts became more or less incomprehensible, and do not want that this happens with tek4010.

Otherwise, if anybody has a better proposal, please let me know.

**Comment by larsbrinkhoff** [comment]
I agree this is a bit of a nuisance.  I have this automatically built as a submodule of the [ITS repository](http://github.com/PDP-10/its), so I had to add a workaround to first remove the binary.

**Comment by LegalizeAdulthood** [comment]
Why not use the ["releases" feature of github](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository)?  Then you can make a single executable available or a ZIP file or whatever.

**Comment by rricharz** [comment]
Binaries removed in latest version
